src/converter/one/convert/JfrToOtlp.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package one.convert;
+
+import one.jfr.JfrReader;
+import one.jfr.StackTrace;
+import one.jfr.event.Event;
+import one.proto.Proto;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Converts .jfr output to OpenTelemetry protocol. */
+public class JfrToOtlp extends JfrConverter {
+    private final Index<String> stringPool = new Index<>(String.class, "");
+    private final Index<String> functionPool = new Index<>(String.class, "");
+    private final Proto otlpProto = new Proto(1024);
+
+    private final int resourceProfilesMark;
+    private final int scopeProfilesMark;
+
+    public JfrToOtlp(JfrReader jfr, Arguments args) {
+        super(jfr, args);
+
+        resourceProfilesMark = otlpProto.startField(Otlp.ProfilesData.RESOURCE_PROFILES);
+        scopeProfilesMark = otlpProto.startField(Otlp.ResourceProfiles.SCOPE_PROFILES);
+    }
+
+    @Override
+    protected void convertChunk() {
+        int profileMark = otlpProto.startField(Otlp.ScopeProfiles.PROFILES);
+
+        writeSampleTypes();
+        writeTimingInformation();
+        otlpProto.field(Otlp.Profile.ORIGINAL_PAYLOAD, jfr.currentChunk());
```

Comment:
That kills one of the benefits of the format - compact representation.
I can't imagine a case where we may need JFR embedded in protobuf.